xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/WCAGContext.java

```diff
@@ -171,7 +171,7 @@ private WCAGTestResults(String testMethodName, String url, String pageClassName,
                 // the default behavior will be to add it to the fails.
                 // In order to resolve these test-suite fails quickly, set them as "false" in FAILS_ON_RULE.
                 .collect(Collectors.toList());
-            this.failAmount = failingViolations.size();
+            this.failAmount = failingViolations.stream().mapToInt(rule -> rule.getNodes().size()).sum();
```

Comment:
Even though it a simple logic, wdyt of factorizing this with the code of line 184?